chef-config/lib/chef-config/config.rb

```diff
@@ -607,7 +623,15 @@ def self.init_openssl
     # `node_name` of the client.
     #
     # If chef-zero is enabled, this defaults to nil (no authentication).
-    default(:client_key) { chef_zero.enabled ? nil : platform_specific_path("/etc/chef/client.pem") }
+    default(:client_key) do
+      if chef_zero.enabled
+        nil
+      elsif target_mode?
+        platform_specific_path("/etc/chef/#{target_mode.host}/client.pem")
+      else
+        platform_specific_path("/etc/chef/client.pem")
+      end
+    end
```

Comment:
so i forgot that the credentials stuff was all knife-side and this is going to be entirely client-side.

it still feels like we should probably support credentials in the client here to have one way of doing things, rather than two.